tests/jvm/src/test/scala/scala/meta/tests/trees/ParentSuite.scala

```diff
@@ -0,0 +1,26 @@
+package scala.meta
+package tests
+package trees
+
+import org.scalatest._
+
+class ParentSuite extends FunSuite {
+
+  test("Tree.transform does not preserve parent chain origins") {
```

Comment:
Can we add more detailed tests?

- validate only `Lit.Int(3)` is missing pos
- validate all other nodes have same position as their corresponding node have before